vitrina/datasets/forms.py

```diff
@@ -55,14 +62,17 @@ def __init__(self, *args, **kwargs):
             Submit('submit', button, css_class='button is-primary')
         )
 
-        instance = self.instance if self.instance and self.instance.pk else None
-        if not instance:
+        if not project_instance:
             if Licence.objects.filter(is_default=True).exists():
                 default_licence = Licence.objects.filter(is_default=True).first()
                 self.initial['licence'] = default_licence
             if Frequency.objects.filter(is_default=True).exists():
                 default_frequency = Frequency.objects.filter(is_default=True).first()
                 self.initial['frequency'] = default_frequency
+        else:
+            if DatasetGroup.objects.filter(dataset=project_instance).exists():
+                groups = DatasetGroup.objects.filter(dataset=project_instance).all()
```

Comment:
Queries are expensive, so instead of querying if groups exists and then getting all groups, maybe, just get all the groups and check if result is not empty? This way you will save one query.